### ISSUE TITLE
Unable to Add Schedule to Job Template

### ISSUE THREAD

**Issue by iyurev** [ansible/awx#164]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

 - UI
##### SUMMARY

When I try add sheduler to job template don't working 
interctive buttons **START DATE, LOCAL TIME ZONE, REPEATE FREQUENCY**
LOCAL TIME ZONE and  REPEATE FREQUENCY  - not responce, click on   START DATE  refresh page. 

##### STEPS TO REPRODUCE

Add  shedule to template.

##### EXPECTED RESULTS

Normal  shedule to template .

##### ACTUAL RESULTS

Menu of user interface for adding  shedule  not working currect

**Comment by matburt** [comment]
Please provide a screenshot and any errors from your browser's javascript console

**Comment by iyurev** [comment]
Below error from my javascript console when I click on calendar icon located under **START DATE**  item:

```
vendor.c7c34fcde5e8a885ab9c.js:438 TypeError: Cannot read property 'schedulerName' of undefined
    at Object.isValid (vendor.c7c34fcde5e8a885ab9c.js:451)
    at vendor.c7c34fcde5e8a885ab9c.js:451
    at Scope.$broadcast (vendor.c7c34fcde5e8a885ab9c.js:438)
    at app.c7c34fcde5e8a885ab9c.js:404
    at ChildScope.$emit (vendor.c7c34fcde5e8a885ab9c.js:438)
    at app.c7c34fcde5e8a885ab9c.js:404
    at Scope.$digest (vendor.c7c34fcde5e8a885ab9c.js:438)
    at Scope.$apply (vendor.c7c34fcde5e8a885ab9c.js:438)
    at done (vendor.c7c34fcde5e8a885ab9c.js:438)
    at completeRequest (vendor.c7c34fcde5e8a885ab9c.js:438)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$broadcast @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ app.c7c34fcde5e8a885ab9c.js:404
$emit @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ app.c7c34fcde5e8a885ab9c.js:404
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
invoke @ vendor.c7c34fcde5e8a885ab9c.js:438
doBootstrap @ vendor.c7c34fcde5e8a885ab9c.js:438
bootstrap @ vendor.c7c34fcde5e8a885ab9c.js:438
angularInit @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
fire @ vendor.c7c34fcde5e8a885ab9c.js:399
fireWith @ vendor.c7c34fcde5e8a885ab9c.js:399
ready @ vendor.c7c34fcde5e8a885ab9c.js:399
completed @ vendor.c7c34fcde5e8a885ab9c.js:389
vendor.c7c34fcde5e8a885ab9c.js:438 TypeError: Cannot read property 'schedulerName' of undefined
    at Object.isValid (vendor.c7c34fcde5e8a885ab9c.js:451)
    at vendor.c7c34fcde5e8a885ab9c.js:451
    at Scope.$broadcast (vendor.c7c34fcde5e8a885ab9c.js:438)
    at app.c7c34fcde5e8a885ab9c.js:404
    at ChildScope.$emit (vendor.c7c34fcde5e8a885ab9c.js:438)
    at app.c7c34fcde5e8a885ab9c.js:404
    at watchGroupAction (vendor.c7c34fcde5e8a885ab9c.js:438)
    at ChildScope.$eval (vendor.c7c34fcde5e8a885ab9c.js:438)
    at Scope.$digest (vendor.c7c34fcde5e8a885ab9c.js:438)
    at Scope.$apply (vendor.c7c34fcde5e8a885ab9c.js:438)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$broadcast @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ app.c7c34fcde5e8a885ab9c.js:404
$emit @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ app.c7c34fcde5e8a885ab9c.js:404
watchGroupAction @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
done @ vendor.c7c34fcde5e8a885ab9c.js:438
completeRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
xhr.onload @ vendor.c7c34fcde5e8a885ab9c.js:438
XMLHttpRequest.send (async)
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
sendReq @ vendor.c7c34fcde5e8a885ab9c.js:438
serverRequest @ vendor.c7c34fcde5e8a885ab9c.js:438
processQueue @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
$eval @ vendor.c7c34fcde5e8a885ab9c.js:438
$digest @ vendor.c7c34fcde5e8a885ab9c.js:438
$apply @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
invoke @ vendor.c7c34fcde5e8a885ab9c.js:438
doBootstrap @ vendor.c7c34fcde5e8a885ab9c.js:438
bootstrap @ vendor.c7c34fcde5e8a885ab9c.js:438
angularInit @ vendor.c7c34fcde5e8a885ab9c.js:438
(anonymous) @ vendor.c7c34fcde5e8a885ab9c.js:438
fire @ vendor.c7c34fcde5e8a885ab9c.js:399
fireWith @ vendor.c7c34fcde5e8a885ab9c.js:399
ready @ vendor.c7c34fcde5e8a885ab9c.js:399
completed @ vendor.c7c34fcde5e8a885ab9c.js:389
2vendor.c7c34fcde5e8a885ab9c.js:438 TypeError: $(...).tooltip is not a function
    at link (http://some_awx_url/static/js/app.c7c34fcde5e8a885ab9c.js:404:415194)
    at invokeLinkFn (http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:70399)
    at nodeLinkFn (http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:55720)
    at compositeLinkFn (http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:45981)
    at compositeLinkFn (http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:46071)
    at http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:44694
    at http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:47321
    at controllersBoundTransclude (http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:53173)
    at http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:236899
    at Scope.$digest (http://some_awx_url/static/js/vendor.c7c34fcde5e8a885ab9c.js:438:129068) <button toolbar-button="" mode="options.mode" aw-tool-tip="{{options.awToolTip}}" data-tip-watch="{{options.dataTipWatch}}" data-placement="{{options.dataPlacement}}" data-container="{{options.dataContainer}}" class="{{options.actionClass}} ng-isolate-scope btn btn-xs btn-primary" data-title="{{options.dataTitle}}" ng-disabled="{{options.ngDisabled}}" ng-click="$eval(options.ngClick)" ng-show="{{options.ngShow}}" toolbar="true" aw-feature="{{options.awFeature}}">
```

**Comment by matburt** [comment]
I edited your comment to apply markdown formatting. Just keep formatting in mind in the future, it helps the brain parse it!  This is also a good candidate for pastebin or gist.

We'll take a look at this asap.

**Comment by ghost** [comment]
Any workaround to suggest in the meantime ? 
If I'm not mistaken, schedules aren't configurable from the API ? (no POST method available), although tower-cli offers the option (Tower CLI 3.1.7) .
Cheers

**Comment by matburt** [comment]
*everything* is available through the api but not at the top level schedules endpoints, you'll need to create it on the thing you are wanting to schedule: `/api/v1/job_templates/<id>/schedules/`

**Comment by dzeban** [comment]
This is the same as #123 but the last one was closed without any reason.

**Comment by jlmitch5** [comment]
@leifmadsen @dzeban @flabatut @badeveloper 

Just wanted to provide some clarification here as things have been a bit confusing, and I apologize about that!

A few days ago @mabashian tested this issue in our Tower release branch to see if he could see this issue.  He did not see it, so the assumption was that this fix would land when that code was merged back into the devel branch.

Unfortunately, this issue seems to actually be specific to new (ui build-process) code being added into AWX (and not present in our release branch of tower) so the assumption that this fix would flow up from downstream wasn't correct in this particular instance.

I got a traceback when going to the ui route `#/templates/job_template/5/schedules/add`:

```
TypeError: Cannot read property 'amd' of undefined
```

(As far as I understand it) AMD is a type of JS module spec that was created before the ES6 module spec was added.  Our changes to the build process involved disallowing these types of modules because they were getting preferred by webpack and making a muck of other dependencies that needed to be loaded.  (EDIT: we didn't realize that this would break our scheduler dependency which, at the current moment, is an AMD-based module)

In short, @gconsidine who did a lot of work in making our build process better is gonna take a look at this one (which will either be incoming as a PR to awx/devel or to our scheduler library which is abstracted and lives in it's own repo, not flowing from our downstream tower work).

We will ping here once that fix happens.

**Comment by dzeban** [comment]
@jlmitch5 Thanks for the clarification! I'll gladly test the fix in my environment!

**Comment by leifmadsen** [comment]
Yep will test once available. Now you know why I didn't like the workflow of closing my issue #123 :)

**Comment by dougfultz** [comment]
For me the page refreshes, then the above is shown in the browser console.  When the page refreshes it removes anything previously entered into the fields and shows this error:
````
The scheduler options are invalid or incomplete.
````

**Comment by jlmitch5** [comment]
@gconsidine can this be closed?

**Comment by gconsidine** [comment]
@jlmitch5 yup, this is resolved after #244 was merged -- #245 ended up being unrelated though, so that should stay open. @mabashian has a fix in the works for that.
